### PR TITLE
Auto some sdn-1282 EgressIP cases

### PR DESCRIPTION
**Pull request by huiran0826**
Auto OCP-40933 OCP-40928 OCP-40957 OCP-40956

Logs:
https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/Runner-v3-smoke/3115/
https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/Runner-v3-smoke/3116/
https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/Runner-v3-smoke/3117/
https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/Runner-v3-smoke/3118/

/cc @openshift/team-sdn-qe

### CODE REVIEW

File: features/networking/egress-ip.feature [hunk 1]
```diff
@@ -582,3 +586,195 @@ Feature: Egress IP related features
...
+  # @author huirwang@redhat.com
+  # @case_id OCP-40957
+  @admin
+  @destructive
+  Scenario: [SDN-1282] Auto EgressIPs assignments: if a pod is not on a node hosting an egressIP it is random which egressIP it will use
+    Given I save ipecho url to the clipboard
+    Given I store the masters in the :masters clipboard
+    Given I store the schedulable workers in the :workers clipboard
+    Given I store a random unused IP address from the reserved range to the clipboard
+    And evaluation of `IPAddr.new("<%= cb.subnet_range %>").to_s+"/"+IPAddr.new("<%= cb.subnet_range %>").prefix.to_s` is stored in the :valid_subnet clipboard
```
Comment (rbbratta) [inline features/networking/egress-ip.feature:686]: Not necessary but we could use a lambda to do two operation on the same intermediate result.

```ruby
lambda { |i| "#{i.to_s}/#{i.prefix.to_s}" }.call(IPAddr.new("<%= cb.subnet_range %>"))

```

Comment (huiran0826) [reply features/networking/egress-ip.feature]: Good idea, thanks Ross, updated it.

File: features/networking/egress-ip.feature [hunk 2]
```diff
@@ -582,3 +586,195 @@ Feature: Egress IP related features
...
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+
+    # patch two egress ips to new project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ips[0] %>","<%= cb.valid_ips[1] %>"]} |
+
+    # Check the source IP is pods's located node's egress ip
+    When I execute on the pod:
+      | bash | -c | for i in `seq 1 10`; do curl -s --connect-timeout 2 <%= cb.ipecho_url %> ; sleep 2;echo ""; done;  |
```
[the remark this answers is not in the document]
Comment (huiran0826) [reply features/networking/egress-ip.feature]: Updated it, thanks!

File: features/networking/egress-ip.feature [hunk 3]
```diff
@@ -582,3 +586,195 @@ Feature: Egress IP related features
...
+    When I run oc create over "list_for_pods.json" replacing paths:
+      | ["items"][0]["spec"]["template"]["spec"]["nodeName"] | <%= cb.nodes[1].name %> |
+      | ["items"][0]["spec"]["replicas"]                     | 1                       |
+    Then the step should succeed
+    Given a pod becomes ready with labels:
+      | name=test-pods |
+
+    # patch two egress ips to new project
+    Given as admin I successfully merge patch resource "netnamespace/<%= project.name %>" with:
+      | {"egressIPs": ["<%= cb.valid_ips[0] %>","<%= cb.valid_ips[1] %>"]} |
```
[the remark this answers is not in the document]
Comment (huiran0826) [reply features/networking/egress-ip.feature]: Thanks Ross, it is more concise, updated it.

File: features/networking/egress-ip.feature [hunk 4]
```diff
@@ -130,6 +130,8 @@ Feature: Egress IP related features
     """
     as admin I successfully merge patch resource "hostsubnet/<%= node.name %>" with:
       | {"egressCIDRs":null}   |
+    as admin I successfully merge patch resource "hostsubnet/<%= node.name %>" with:
+      | {"egressIPs":null}   |
```
Comment (liangxia) [inline features/networking/egress-ip.feature:134]: Is it possible to do the two patches in one command ?

Comment (huiran0826) [reply features/networking/egress-ip.feature:134]: Thanks for reviewing, no, it cannot be done in one command.

**Comment by huiran0826** [comment]
@liangxia Could you help to merge this PR? Thanks!

**Pull request merged**